src/modules/git_commit.rs

```diff
@@ -0,0 +1,52 @@
+use super::{Context, Module, RootModuleConfig};
+use git2::Repository;
+
+use crate::configs::git_commit::GitCommitConfig;
+
+pub fn id_to_hex_abbrev(bytes: &[u8], len: usize) -> String {
+    bytes
+        .iter()
+        .map(|b| format!("{:02x}", b))
+        .take(len)
+        .collect::<Vec<String>>()
+        .join("")
+}
+
+/// Creates a module with the Git commit in the current directory
+///
+/// Will display the commit hash if the current directory is a git repo
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("git_commit");
+    let config = GitCommitConfig::try_load(module.config);
+    module
+        .get_prefix()
+        .set_value(config.prefix)
+        .set_style(config.style);
+    module
+        .get_suffix()
+        .set_value(config.suffix)
+        .set_style(config.style);
+    module.set_style(config.style);
+
+    let repo = context.get_repo().ok()?;
+    let repo_root = repo.root.as_ref()?;
+    let git_repo = Repository::open(repo_root).ok()?;
+
+    let git_head = git_repo.head();
+    if let Ok(git_head) = git_head {
+        let head_commit = git_head.peel_to_commit().ok()?;
+        let commit_oid = head_commit.id();
+        module.create_segment(
+            "hash",
+            &config
+                .hash
+                .with_value(&id_to_hex_abbrev(commit_oid.as_bytes(), 7)),
```

Comment:
I find the hash to be rather long. I think convention for the abbreviated commit is for it to be 7 characters long, such as when you run `git log --abbrev-commit`.

Right now it looks like it's coming out to 14 characters.
What are your thoughts around shortening it further?

Reply:
> I find the hash to be rather long. I think convention for the abbreviated commit is for it to be 7 characters long, such as when you run `git log --abbrev-commit`.
> 
> Right now it looks like it's coming out to 14 characters.
> What are your thoughts around shortening it further?

If I can throw in my two cents, I'd also vote for a 7-character hash. Both GitHub and BitBucket display that length for short commits, and it also takes up less screen real estate in the prompt. 🙂

Reply:
So I did extensive research into this. Initially I was thinking of making it dynamic to match the behavior of `git rev-parse --short HEAD`.

After reading git's source code, it turned out to be a very deep rabbit hole. Git2 doesn't have an API for this. The actual implementation is at https://github.com/git/git/blob/master/sha1-name.c#L670. Which will result in hundreds lines of rust code counting all the helper functions it calls.

I ended up stick with the default length in Git to keep it simple as the first iteration, which is defined as 7: https://github.com/git/git/blob/master/builtin/rev-parse.c#L728. This should be long enough to work for majority of the code base. We can certainly port the dynamic length calculating algorithm from git if we ever get any request for increasing the length.

Reply:
Since each byte is encoded as 2 hexadecimal characters, converting `7` bytes makes for a hash that is 14 characters long. A different approach may be needed to have 3.5 bytes displayed.

Reply:
if we change a different encoding method, the displayed commit hash won't be useful for copy paste anymore. In order to use the displayed commit hash with the git cli, one would have to convert that 3.5 bytes into hex encoding first manually.

If you are still very concerned with the length, how about we make it configurable so users can pick whatever length they are most comfortable with?

Reply:
> if we change to a different encoding method, the displayed commit hash won't be useful for copy paste anymore.

In my workflow, a 7-character hash is as useful as a 14-character hash, but I'm certainly open to the idea of adding a hash truncation configuration value. 👍 

TIL that `git` will allow you to use a hash as small as 4 characters:
![image](https://user-images.githubusercontent.com/4658208/70172740-5763f080-169f-11ea-8f7a-8906f260e64a.png)

Reply:
yes, for smaller git repos, 7 bytes is more than enough, but will run into edge-cases for larger repos.

I have pushed an update to make the commit length configurable. It's named `min_commit_hash_length` to be consistent with git. This also means when we add dynamic length calculation in the future, the naming of this config will still make sense.

Let me know if you have any other thought on the implementation!

Reply:
What would dynamic length calculation entail?
I was thinking that it would be a static value (`commit_hash_length`).

Reply:
They way `git rev-parse --short HEAD` works is it starts with a user defined minimal length, and then traverse all git objects to to see if there is any other object shares the same hash prefix, if so, it increases the length until it becomes a unique prefix.

Reply:
I have renamed it to `commit_hash_length`, we could also just remove this config when dynamic hash length is implemented.